src/requests/features.jl

```diff
@@ -236,17 +210,13 @@ function textDocument_range_formatting_request(params::DocumentRangeFormattingPa
         )
     end
 
-    if longest_prefix !== nothing && !isempty(longest_prefix)
-        io = IOBuffer()
-        for line in eachline(IOBuffer(newcontent), keep=true)
-            print(io, longest_prefix, line)
-        end
-        newcontent = String(take!(io))
+    range_regex = Regex("$startmark\n((?s).*)\n\\s*$stopmark")
+    range_formatted = match(range_regex, text_formatted)
+    range_unformatted = match(range_regex, text_marked)
+    if isnothing(range_formatted) || (range_formatted[1] == range_unformatted[1])
+        return TextEdit[]
     end
-
-    lsedits = TextEdit[TextEdit(Range(l1, c1, l2, c2), newcontent)]
-
-    return lsedits
+    return TextEdit[TextEdit(Range(params.range.start.line, 0, params.range.stop.line, 99999999), range_formatted[1])]
```

Comment:
You can get the end char as a return value from `mark_range` (`sizeof(textlines[stopline])`)

Reply:
`mark_range` does not have access to the formatted range.
What about getting it from the `range_formatted` text like this: `sizeof(last(split(range_formatted[1], "\n")))`

Reply:
We don't care about the formatted text. This is a range into the unformatted text that's replaced.

Reply:
or `sizeof(range_unformatted[1]) - findlast('\n', range_unformatted[1])`

Reply:
Ah, got it.

Reply:
Ahhh got it.. it should be `sizeof(textlines[stopline + 1])`